src/table.js

```diff
@@ -617,34 +617,38 @@ export function coerceToType(value, type) {
   }
 }
 
+export function getSchema(source) {
+  const {columns} = source;
+  let {schema} = source;
+  if (!isQueryResultSetSchema(schema)) {
+    schema = inferSchema(source, isQueryResultSetColumns(columns) ? columns : undefined);
+    return {schema, shouldCoerce: true};
+  }
+  return {schema, shouldCoerce: false};
```

Comment:
Maybe we should call this “inferred” instead of “shouldCoerce”? Because “inferred” makes a factual observation about how the schema was derived, whereas “shouldCoerce” implies that the caller should do something with the returned schema—and perhaps the caller wants to do something else?

Reply:
good call, updated.